OMEdit/OMEditGUI/Editors/CompositeModelEditor.cpp

```diff
@@ -528,9 +534,15 @@ void CompositeModelEditor::updateConnection(LineAnnotation *pConnectionLineAnnot
     if (connection.attribute("From").compare(pConnectionLineAnnotation->getStartComponentName()) == 0 &&
         connection.attribute("To").compare(pConnectionLineAnnotation->getEndComponentName()) == 0) {
       connection.setAttribute("Delay", pConnectionLineAnnotation->getDelay());
-      connection.setAttribute("alpha", pConnectionLineAnnotation->getAlpha());
-      connection.setAttribute("Zf", pConnectionLineAnnotation->getZf());
-      connection.setAttribute("Zfr", pConnectionLineAnnotation->getZfr());
+      if(!pConnectionLineAnnotation->getAlpha().isEmpty()) {
+        connection.setAttribute("alpha", pConnectionLineAnnotation->getAlpha());
+      }
+      if(!pConnectionLineAnnotation->getZf().isEmpty()) {
+        connection.setAttribute("Zf", pConnectionLineAnnotation->getZf());
+      }
+      if(!pConnectionLineAnnotation->getZfr().isEmpty()) {
+        connection.setAttribute("Zfr", pConnectionLineAnnotation->getZfr());
+      }
```

Comment:
That's true, if the user writes an empty value in the dialog it would keep the old value. I changed it to "hasAttribute" instead of "isEmpty", so we never create new attributes when applying changes in the dialog. When creating new connections, however, we need to use isEmpty, to avoid creating the irrelevant attributes in the first place.